src/plugins/point_cloud/PointCloud.hh

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * Development of this module has been funded by the Monterey Bay Aquarium
+ * Research Institute (MBARI) and the David and Lucile Packard Foundation
+ */
+
+#ifndef IGNITION_GUI_PLUGINS_POINTCLOUD_HH_
+#define IGNITION_GUI_PLUGINS_POINTCLOUD_HH_
+
+#include <memory>
+
+#include <ignition/msgs/float_v.pb.h>
+#include <ignition/msgs/pointcloud_packed.pb.h>
+
+#include "ignition/gui/Plugin.hh"
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  class PointCloudPrivate;
+
+  /// \brief Visualize PointCloudPacked messages in the 3D scene.
+  /// By default, the whole cloud is displayed using a single color. Users
+  /// can optionally choose a topic publishing FloatV messages which will be
+  /// used to color all points with a color gradient according to their values.
```

Comment:
A bit more guidelines about the FloatV message should be given - its index has to match the point cloud (and what happens if it doesn't, undefined behavior or error), what kind of values (in range [0, 1] or whatever, I know it's determined automatically, but it's not clear to the reader).

Reply:
Added, let me know what you think.

Reply:
Nice! Example looks good